crates/iota-rpc-loadgen/src/main.rs

```diff
@@ -27,115 +27,115 @@ use crate::{
+    #[arg(long, num_args(1..), default_value = "http://127.0.0.1:9000")]
     pub urls: Vec<String>,
     /// the path to log file directory
-    #[clap(long, default_value = "~/.iota/iota_config/logs")]
+    #[arg(long, default_value = "~/.iota/iota_config/logs")]
     logs_directory: String,
 
-    #[clap(long, default_value = "~/.iota/loadgen/data")]
+    #[arg(long, default_value = "~/.iota/loadgen/data")]
     data_directory: String,
 }
 
 #[derive(Parser)]
 pub struct CommonOptions {
-    #[clap(short, long, default_value_t = 0)]
+    #[arg(short, long, default_value_t = 0)]
     pub repeat: usize,
 
-    #[clap(short, long, default_value_t = 0)]
+    #[arg(short, long, default_value_t = 0)]
     pub interval_in_ms: u64,
 
     /// different chunks will be executed concurrently on the same thread
-    #[clap(long, default_value_t = 1)]
+    #[arg(long, default_value_t = 1)]
     num_chunks_per_thread: usize,
 }
 
 #[derive(Parser)]
 pub enum ClapCommand {
-    #[clap(name = "dry-run")]
+    #[command(name = "dry-run")]
     DryRun {
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
     },
-    #[clap(name = "get-checkpoints")]
+    #[command(name = "get-checkpoints")]
     GetCheckpoints {
         /// Default to start from checkpoint 0
-        #[clap(short, long, default_value_t = 0)]
+        #[arg(short, long, default_value_t = 0)]
         start: u64,
 
         /// inclusive, uses `getLatestCheckpointSequenceNumber` if `None`
-        #[clap(short, long)]
+        #[arg(short, long)]
         end: Option<u64>,
 
-        #[clap(long)]
+        #[arg(long)]
         skip_verify_transactions: bool,
 
-        #[clap(long)]
+        #[arg(long)]
         skip_verify_objects: bool,
 
         // Whether to record data from checkpoint
-        #[clap(long)]
+        #[arg(long)]
         skip_record: bool,
 
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
     },
-    #[clap(name = "pay-iota")]
+    #[command(name = "pay-iota")]
     PayIota {
         // TODO(chris) customize recipients and amounts
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
     },
-    #[clap(name = "query-transaction-blocks")]
+    #[command(name = "query-transaction-blocks")]
     QueryTransactionBlocks {
-        #[clap(long, ignore_case = true)]
+        #[arg(long, ignore_case = true)]
         address_type: AddressQueryType,
 
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
     },
-    #[clap(name = "multi-get-transaction-blocks")]
+    #[command(name = "multi-get-transaction-blocks")]
     MultiGetTransactionBlocks {
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
     },
-    #[clap(name = "multi-get-objects")]
+    #[command(name = "multi-get-objects")]
     MultiGetObjects {
-        #[clap(flatten)]
+        #[command(flatten)]
         common: CommonOptions,
```

Comment:
CommonOptions is not an arg?

Reply:
Or does it only work for basic types and structs need the `command`?

Reply:
It's a set of args, but no it's not an arg itself